library/spdm_requester_lib/libspdm_req_send_receive.c

```diff
@@ -207,6 +232,229 @@ libspdm_return_t libspdm_receive_response(void *context, const uint32_t *session
+        spdm_request_size = 0;
+
+        if (LIBSPDM_STATUS_IS_ERROR(status)) {
+            break;
+        }
+
+        response = message;
+        response_size = message_size;
+
+        libspdm_zero_mem(response, response_size);
+
+        status = libspdm_receive_response(
+            spdm_context, session_id, false,
+            &response_size, &response);
+
+        if (LIBSPDM_STATUS_IS_ERROR(status)) {
+            break;
+        }
+        spdm_response = (void*) (message);
+
+        if (response_size < sizeof(spdm_message_header_t)) {
+            status = LIBSPDM_STATUS_INVALID_MSG_SIZE;
+            break;
+        }
+
+        if (spdm_response->header.request_response_code == SPDM_ERROR
+            && spdm_response->header.param1 == SPDM_ERROR_CODE_LARGE_RESPONSE) {
+
+            /* It is possible that the CHUNK_SEND_ACK + chunk response is larger
+             * than the DATA_TRANSFER_SIZE. In this case an ERROR_LARGE_RESPONSE
+             * is returned directly in the response buffer rather than part of
+             * the CHUNK_SEND_ACK. Store this error response in scratch buffer
+             * to be handled when reading response */
+            if (response_size < send_info->large_message_capacity) {
+                libspdm_copy_mem(
+                    send_info->large_message, send_info->large_message_capacity,
+                    spdm_response, response_size);
+                send_info->large_message_size = response_size;
+            }
+            else {
+                status = LIBSPDM_STATUS_INVALID_MSG_SIZE;
+                break;
+            }
+        }
+        else {
+
+            if (spdm_response->header.spdm_version
+                != libspdm_get_connection_version(spdm_context)) {
+                status = LIBSPDM_STATUS_INVALID_MSG_FIELD;
+                break;
+            }
+            if (spdm_response->header.request_response_code != SPDM_CHUNK_SEND_ACK) {
+                status = LIBSPDM_STATUS_INVALID_MSG_FIELD;
+                break;
+            }
+
+            if (response_size < sizeof(spdm_chunk_send_ack_response_t)) {
+                status = LIBSPDM_STATUS_INVALID_MSG_SIZE;
+                break;
+            }
+            if (spdm_response->header.param1
+                & SPDM_CHUNK_SEND_ACK_RESPONSE_ATTRIBUTE_EARLY_ERROR_DETECTED) {
+                status = LIBSPDM_STATUS_INVALID_MSG_FIELD;
+                break;
+            }
+            if (spdm_response->header.param2 != send_info->chunk_handle) {
+                status = LIBSPDM_STATUS_INVALID_MSG_SIZE;
+                break;
+            }
+            if (send_info->chunk_seq_no != spdm_response->chunk_seq_no) {
+                status = LIBSPDM_STATUS_INVALID_MSG_FIELD;
+                break;
+            }
+            if ((send_info->chunk_bytes_transferred >= send_info->large_message_size
+                 && !(spdm_response->header.param1 & SPDM_CHUNK_SEND_REQUEST_ATTRIBUTE_LAST_CHUNK))
```

Comment:
What's going on here? `CHUNK_SEND_ACK.Param1.LastChunk` doesn't exist.

Reply:
I'm detecting the error case where we have a bad responder that is
1) Not setting the LastChunk when All bytes have been sent
Or 
2) Setting the LastChunk bit when Not All bytes have been sent

Reply:
The Requester sets `CHUNK_SEND.Param1.LastChunk`. The check should be done by the Responder.

Reply:
Ah! Right.